src/modules/apps/link.ts

```diff
@@ -29,7 +31,71 @@ const UPDATE_SIGN = chalk.blue('U')
 const stabilityThreshold = process.platform === 'darwin' ? 100 : 200
 const AVAILABILITY_TIMEOUT = 1000
 const N_HOSTS = 3
+const builderHubInjectedDepsTimeout = 2000  // 2 seconds
+const yarn = join(__dirname, '../../../node_modules/yarn/bin/yarn.js install --force')
+
+const resolvePackageJsonPath = (builder: string) => resolvePath(process.cwd(), `${builder}/package.json`)
+
+const resolveAppId = async (appName: string, appVersion: string) => await apps.getApp(`${appName}@${appVersion}`).then(prop('id'))
```

Comment:
this could be moved to `modules/apps/utils.ts`

Reply:
Done